nucleus/metrics/segmentation_metrics.py

```diff
@@ -141,52 +192,47 @@ def __init__(
                 (AND), forming a more selective `and` multiple field predicate.
                 Finally, the most outer list combines these filters as a disjunction (OR).
         """
-        assert (
-            0 <= iou_threshold <= 1
-        ), "IoU threshold must be between 0 and 1."
-        self.iou_threshold = iou_threshold
         super().__init__(
-            enforce_label_match,
-            confidence_threshold,
             annotation_filters,
             prediction_filters,
         )
-        self.metric = PolygonIOU(
-            self.enforce_label_match,
-            self.iou_threshold,
-            self.confidence_threshold,
-            self.annotation_filters,
-            self.prediction_filters,
+
+    def _metric_impl(
+        self,
+        annotation_img: np.ndarray,
+        prediction_img: np.ndarray,
+        annotation: SegmentationAnnotation,
+        prediction: SegmentationPrediction,
+    ) -> ScalarResult:
+        confusion = self._calculate_confusion_matrix(
+            annotation, annotation_img, prediction, prediction_img
         )
 
-    def call_poly_metric(
-        self, annotations: AnnotationList, predictions: PredictionList
-    ):
-        return self.metric(annotations, predictions)
+        with np.errstate(divide="ignore", invalid="ignore"):
+            iou = np.diag(confusion) / (
+                confusion.sum(axis=1)
+                + confusion.sum(axis=0)
+                - np.diag(confusion)
+            )
+        return ScalarResult(value=np.nanmean(iou), weight=annotation_img.size)  # type: ignore
```

Comment:
Correct me if I'm wrong, but this function reads to me as a pixel wise iou computation (i.e. there's only pixel-level matching, no instance level matching). As such, it seems unexpected to me that this is only matching by index and not by label. AKA imagine someone passes an instance segmentation mask into this function, and the segments in the GT mask are:

```
{
    "car": 1,
    "truck": 2,
    "grass": 3,
}
```
and the predicted mask has segments:
```
{
    "car": 1,
    "truck": 2,
    "grass": 3,
    "car": 4,
}
```

I think I would expect pixel locations with gt 1 and pred 4 to be considered a "match" .


Same argument applies for when the indices in mask vs. GT are ordered differently, i.e. pred mask:
```
{
    "car": 2,
    "truck": 1,
    "grass": 4,
    "car": 3,
}
```

Reply:
sounds good, I am happy to leave that as a TODO